src/test/java/pl/rejmanbeata/bakery/controller/client/ClientControllerE2ETest.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+@Transactional
+class ClientControllerE2ETest {
+    public static final String CLIENTS_PATH = "/clients";
+    public static final String CLIENTS_ID_PATH = "/clients/{id}";
+    public static final int DB_SIZE = 3;
+    public static final long ID = 1L;
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Test
+    void shouldGetClientById_Found() throws Exception {
+        mockMvc.perform(get(CLIENTS_ID_PATH, ID)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.lastName").value("Doe"))
+                .andExpect(jsonPath("$.name").value("John"));
+    }
+
+    @Test
+    void shouldGetClientById_NotFound() throws Exception {
+        mockMvc.perform(get(CLIENTS_ID_PATH, 76L)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    void shouldCreateClient() throws Exception {
+        mockMvc.perform(get(CLIENTS_PATH)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", Matchers.hasSize(DB_SIZE)));
+
+        String body = "{\"name\":\"Harry\",\"lastName\":\"Potter\",\"address\":{\"longitude\":12.4924,\"latitude\":41.8902}}";
+
+        mockMvc.perform(post(CLIENTS_PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(body))
+                .andDo(print())
+                .andExpect(status().isCreated())
+                .andExpect(jsonPath("$.name").value("Harry"));
+
+        mockMvc.perform(get(CLIENTS_PATH)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", Matchers.hasSize(DB_SIZE + 1)));
+    }
+
+    @Test
+    void shouldDeleteClient_Found() throws Exception {
+        mockMvc.perform(get(CLIENTS_PATH)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", Matchers.hasSize(DB_SIZE)));
+
+        mockMvc.perform(delete("/clients/{id}", ID))
+                .andExpect(status().isNoContent());
+
+        mockMvc.perform(get(CLIENTS_PATH)
```

Comment:
we can check that as it is or
we can check if asking for that particular client results in 404 :) 

BTW: why are Clients test included in Employees PR :) ?